server/sockets.js

```diff
@@ -1,12 +1,24 @@
 const socketio = require('socket.io');
+const socketManager = require('./socket-manager.js');
+const cookieParser = require('cookie');
 
-module.exports =  (listener) => {
+const createSocket = (listener) => {
   const io = socketio.listen(listener);
-  io.on('connection', (socket) => {
-    console.log("new connection");
+  
+  io.set('authorization', (handshakeData, accept) => {
 
-    socket.on('newGoals', (data) => {
-      console.log(data);
-    });
+    if (handshakeData.headers.cookie) {
+      const cookie = cookieParser.parse(handshakeData.headers.cookie)['new-user'];
```

Comment:
Okay I think you are really close.

I get that maybe your old system was not working with this. But it seems that you solution to this was to implement 2 authentication systems on the server. Is this correct? This does not seem like an optimal solution to your problem